pkg/rbd/controllerserver.go

```diff
@@ -462,6 +462,48 @@ func storeSnapshotMetadata(rbdSnap *rbdSnapshot, cp util.CachePersister) error {
 	return nil
 }
 
+// ExpandVolume expand RBD Volumes on demand based on resizer request
+func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
+
+	if err := cs.Driver.ValidateControllerServiceRequest(csi.ControllerServiceCapability_RPC_EXPAND_VOLUME); err != nil {
+		klog.Warningf("invalid expand volume req: %v", protosanitizer.StripSecrets(req))
+		return nil, err
+	}
+
+	volID := req.GetVolumeId()
+	if volID == "" {
+		return status.Error(codes.InvalidArgument, "Volume ID cannot be empty")
+	}
+
+	capRange := req.GetCapacityRange()
+	if capRange == nil {
+		return status.Error(codes.InvalidArgument, "CapacityRange cannot be empty")
+	}
+
+	secret := req.GetSecrets()
+
+	rbdVolume, err := getRBDVolumeByID(volID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Source Volume ID %s cannot found", req.GetSourceVolumeId())
+	}
+
+	//resize volume if required
+	if (rbdVolume.VolSize) < capRange.CapacityBytes {
```

Comment:
Is ```VolSize``` stored in MiB? I ask since it's treated like it's stored in bytes here but treated like it's stored in MiB on line 568

Reply:
You are correct, its on MiB. I will convert the CapacityBytes to MiB.

Reply:
Both are int64 @dillaman , so we should be good here.